src/main/java/io/stargate/sgv2/jsonapi/service/shredding/Shredder.java

```diff
@@ -403,36 +384,61 @@ private void validateDocDepth(DocumentLimitsConfig limits, int depth) {
     }
   }
 
-  static class ValueValidator {
+  /**
+   * Secondary validator applied to the storable document after non-indexable fields (and branches)
+   * have been pruned.
+   */
+  static class IndexableValueValidator {
     final DocumentLimitsConfig limits;
 
-    public ValueValidator(DocumentLimitsConfig limits) {
+    public IndexableValueValidator(DocumentLimitsConfig limits) {
       this.limits = limits;
     }
 
     public void validate(ObjectNode doc) {
-      validateObjectValue(doc);
+      validateObjectValue(null, doc);
     }
 
-    private void validateValue(JsonNode value) {
+    private void validateValue(String referringPropertyName, JsonNode value) {
```

Comment:
Need to pass name of property for value, to include in error messages (formerly not needed)